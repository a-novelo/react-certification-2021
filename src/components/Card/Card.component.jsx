import React from 'react';
import { useHistory } from 'react-router-dom';
import {
  CardWrapper,
  CardButton,
  CardImage,
  CardBody,
  CardTitle,
  CardDescription,
} from './Card.styled';

function Card(props) {
  const history = useHistory();
  const { photoHeader, title, description, videoId } = props;

  return (
    <CardWrapper>
      <CardButton
        onClick={() => {
          history.push(`/video/${videoId}`);
        }}
      >
        <CardImage src={photoHeader} />
        <CardBody>
          <CardTitle>{title}</CardTitle>
          <CardDescription>{description}</CardDescription>
        </CardBody>
      </CardButton>
    </CardWrapper>
  );
}

export default Card;
