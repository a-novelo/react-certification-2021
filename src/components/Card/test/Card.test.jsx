import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import Card from '../Card.component';
import SearchProvider from '../../../providers/Search';
import ThemeProvider from '../../../providers/Theme';

const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('#Card', () => {
  it('displays correctly', () => {
    const tree = renderer
      .create(
        <SearchProvider>
          <ThemeProvider>
            <Card />
          </ThemeProvider>
        </SearchProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has button to navigate to video', () => {
    render(
      <SearchProvider>
        <ThemeProvider>
          <Card />
        </ThemeProvider>
      </SearchProvider>
    );
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(mockHistoryPush).toHaveBeenCalledWith('/video/undefined');
  });
});
