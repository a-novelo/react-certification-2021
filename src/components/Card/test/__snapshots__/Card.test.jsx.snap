// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Card displays correctly 1`] = `
<div
  className="sc-bdfBwQ fPgORt"
>
  <button
    className="sc-gsTCUz iSKmng"
    onClick={[Function]}
  >
    <img
      className="sc-dlfnbm OEuyo"
    />
    <div
      className="sc-hKgILt jnTFzE"
    >
      <div
        className="sc-eCssSg enODAG"
      />
      <p
        className="sc-jSgupP jNMAjU"
      />
    </div>
  </button>
</div>
`;
